lib/rmg/modificators/TreasurePlacer.cpp

```diff
@@ -1180,4 +1197,89 @@ void TreasurePlacer::ObjectPool::discardObjectsAboveValue(ui32 value)
 	});
 }
 
+ObjectConfig::EObjectCategory TreasurePlacer::ObjectPool::getObjectCategory(CompoundMapObjectID id)
```

Comment:
Use a map for trivial cases? It will get rid of most of If / else if

Reply:
I'm afraid it will get non-trivial very soon.